src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/AvroHelper.java

```diff
@@ -39,6 +39,8 @@
 /** A helper to convert sql result set to avro format and dump the avro result to output stream. */
 public class AvroHelper {
 
+  private static final Pattern TRAILING_SPACES_REGEX = Pattern.compile("\\s++$");
```

Comment:
Shouldn't this be `\\s+$`?

Reply:
Both are correct. But mine is a bit more optimal \s+ - matches 0 or more, \s++ matches 1 or more. So if there is no space  at the end \s+ will still match and Matcher will replace empty string with another empty string. With \s++ no action is taken in that case.